### PR TITLE
gensio 2.8.5 (new formula) ser2net: update to use gensio formula

### PR DESCRIPTION
**Pull request by cminyard**
- [ x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I submitted this earlier, before gensio had enough likes and forks. It now does. All audits pass.

The ser2net formula builds gensio as part of its build, but gensio is a separate stand-alone library and set of tools. This commit splits gensio into its own separate formula so it can be properly maintained, and makes ser2net depend on that.

As well, the ser2net formula was non-functional. It had the following issues:

* The configuration files were set to inside the /opt/homebrew/etc/ser2net, so every time you upgraded you would lose your configuration. This change moved it to /opt/homebrew/etc/ser2net.

* Same problem with the authentication token location, that is moved to /opt/homebrew/share.

* The service was run in the background, so that caused all kinds of confusion. This switches it to run in the foreground so brew can manage it.

* The administration interface was enabled on the command line. This has security implications and is generally a bad idea outside of testing. The admin interface can be enabled in the configuration files with the proper authentication.

### CODE REVIEW

File: Formula/s/ser2net.rb [hunk 1]
```diff
@@ -20,35 +20,14 @@ class Ser2net < Formula
...
-                            "--with-tcl=no"
-      system "make", "install"
-    end
-
-    ENV.append_path "PKG_CONFIG_PATH", "#{libexec}/gensio/lib/pkgconfig"
-    ENV.append_path "CFLAGS", "-I#{libexec}/gensio/include"
-    ENV.append_path "LDFLAGS", "-L#{libexec}/gensio/lib"
-
     system "./configure", "--disable-dependency-tracking",
                           "--prefix=#{prefix}",
```
Comment (SMillerDev) [inline Formula/s/ser2net.rb:28]:
```suggestion
    system "./configure", *std_configure_args
```

File: Formula/g/gensio.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on "avahi"
+    depends_on "linux-pam"
+  end
+
+  def python3
+    "python3.11"
+  end
+
+  def install
+    # https://rubydoc.brew.sh/Formula.html#std_configure_args-instance_method
```
Comment (SMillerDev) [inline Formula/g/gensio.rb:27]: No need to document what parts of brew a formula is using
```suggestion
```

File: Formula/g/gensio.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    system "./configure", *std_configure_args,
+                          "--disable-silent-rules",
+                          "--with-pythoninstall=#{lib}/gensio-python",
+                          "--sysconfdir=#{etc}",
+                          "--with-tclcflags=-I #{HOMEBREW_PREFIX}/include/tcl-tk"
```
Comment (SMillerDev) [inline Formula/g/gensio.rb:32]: Why not the macOS one?

File: Formula/g/gensio.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  depends_on "tcl-tk"
+  on_linux do
+    depends_on "alsa-lib"
+    depends_on "avahi"
+    depends_on "linux-pam"
+  end
+
+  def python3
+    "python3.11"
+  end
```
Comment (SMillerDev) [inline Formula/g/gensio.rb:24]: Why not the latest one?

File: Formula/g/gensio.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
+class Gensio < Formula
+  desc "Stream I/O Library"
+  homepage "https://github.com/cminyard/gensio"
+  url "https://downloads.sourceforge.net/project/ser2net/ser2net/gensio-2.8.4.tar.gz"
```
Comment (SMillerDev) [inline Formula/g/gensio.rb:4]: Is there no GitHub release?

**Comment by SMillerDev** [comment]
> There is a tag on GitHub, but traditionally I release the tarballs on sourceforge because it's been there a long time and that's where everyone expects it.  I can add it to GitHub, if you like.

If the homepage is GitHub I would expect the code to also come from there. If there is a tag we can just download the tarball from there.

**Comment by cminyard** [comment]
On Tue, Apr 16, 2024 at 06:12:49AM -0700, Sean Molenaar wrote:
> @SMillerDev commented on this pull request.
> 
> 
> 
> >      system "./configure", "--disable-dependency-tracking",
>                            "--prefix=#{prefix}",
> 
> ```suggestion
>     system "./configure", *std_configure_args
> ```

Yes, that's better. That's in the ser2net formula, which I didn't
originally write, so I just didn't change it.

> 
> > +  depends_on ***@***.***"
> +  depends_on "portaudio"
> +  depends_on ***@***.***"
> +  depends_on "tcl-tk"
> +  on_linux do
> +    depends_on "alsa-lib"
> +    depends_on "avahi"
> +    depends_on "linux-pam"
> +  end
> +
> +  def python3
> +    "python3.11"
> +  end
> +
> +  def install
> +    # https://rubydoc.brew.sh/Formula.html#std_configure_args-instance_method
> 
> No need to document what parts of brew a formula is using
> ```suggestion
> ```

Ok, I removed the comment.  I think that came from boilerplate, but I
don't remember.

> 
> > +    depends_on "alsa-lib"
> +    depends_on "avahi"
> +    depends_on "linux-pam"
> +  end
> +
> +  def python3
> +    "python3.11"
> +  end
> +
> +  def install
> +    # https://rubydoc.brew.sh/Formula.html#std_configure_args-instance_method
> +    system "./configure", *std_configure_args,
> +                          "--disable-silent-rules",
> +                          "--with-pythoninstall=#{lib}/gensio-python",
> +                          "--sysconfdir=#{etc}",
> +                          "--with-tclcflags=-I #{HOMEBREW_PREFIX}/include/tcl-tk"
> 
> Why not the macOS one?

I didn't think about that.  It won't compile on Linux without doing
this, so I just used it for both.  I'll fix that.

> 
> > +  def python3
> +    "python3.11"
> +  end
> 
> Why not the latest one?

Oops, yes, that should be fixed.

Thank you,

-corey

File: Formula/g/gensio.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  def python3
+    "python3.12"
+  end
+
+  def install
+    args = %w[--disable-silent-rules]
+    args << "--with-pythoninstall=#{lib}/gensio-python"
+    args << "--sysconfdir=#{etc}"
```
Comment (SMillerDev) [inline Formula/g/gensio.rb:29]:
```suggestion
    args = %W[
      --disable-silent-rules
      --sysconfdir=#{etc}
      --with-pythoninstall=#{lib}/gensio-python
    ]
```

File: Formula/g/gensio.rb [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  depends_on "glib"
+  depends_on "openssl@3"
+  depends_on "portaudio"
+  depends_on "python@3.12"
+  on_linux do
+    depends_on "alsa-lib"
+    depends_on "avahi"
+    depends_on "linux-pam"
+    depends_on "tcl-tk"
+  end
```
Comment (SMillerDev) [inline Formula/g/gensio.rb:20]:
```suggestion
  end
  uses_from_macos "tcl-tk"
```

**Comment by cminyard** [comment]
Ok, one more time.  Hopefully I have everything, I don't see anything else.

**Comment by cminyard** [comment]
Anything else I need to do here?  It's been three weeks without any real comments.

**Review by chenrui333 — APPROVED** [review]
we can followup on the audit failures

**Pull request merged**